pkg/sql/opt/memo/logical_props_builder.go

```diff
@@ -1268,8 +1268,8 @@ func (b *logicalPropsBuilder) buildMutationProps(mutation RelExpr, rel *props.Re
 	// Output Columns
 	// --------------
 	// Only non-mutation columns are output columns.
-	for i, n := 0, tab.ColumnCount(); i < n; i++ {
-		if private.IsColumnOutput(i) {
+	for i, n := 0, tab.AllColumnCount(); i < n; i++ {
+		if !cat.IsMutationColumn(tab, i) && private.IsColumnOutput(i) {
```

Comment:
Actually, I think we can just get rid of the comment and the check for `IsMutationColumn`. `private.IsColumnOutput` should always return false for mutation columns, or else there's a bug somewhere else in the code. We always set `private.ReturnCols` entries to zero for mutation cols.